docs/src/main/asciidoc/resteasy-reactive-migration.adoc

```diff
@@ -0,0 +1,152 @@
+
+|`quarkus-resteasy-jsonb`
+|`quarkus-resteasy-reactive-jsonb`
+
+|`quarkus-resteasy-jaxb`
+|`quarkus-resteasy-reactive-jaxb`
+
+|`quarkus-resteasy-qute`
+|`quarkus-resteasy-reactive-qute`
+
+|===
+
+NOTE: The `quarkus-resteasy-mutiny` does not have a corresponding dependency, as RESTEasy Reactive provides Mutiny integration out of the box.
+
+=== Annotations
+
+RESTEasy Reactive does not support the various custom annotation under the `org.jboss.resteasy.annotations` package.
+
+The following table matches the legacy RESTEasy annotations with the new RESTEasy Reactive ones.
+
+|===
+|Legacy |RESTEasy Reactive |Comments
+
+|`org.jboss.resteasy.annotations.jaxrs.PathParam`
+|`org.jboss.resteasy.reactive.RestPath`
+|This annotation is not necessary when the path part matches the method parameter name
+
+|`org.jboss.resteasy.annotations.jaxrs.QueryParam`
+|`org.jboss.resteasy.reactive.RestQuery`
+|
+
+|`org.jboss.resteasy.annotations.jaxrs.FormParam`
+|`org.jboss.resteasy.reactive.RestForm`
+|
+
+|`org.jboss.resteasy.annotations.jaxrs.HeaderParam`
+|`org.jboss.resteasy.reactive.RestHeader`
+|
+
+|`org.jboss.resteasy.annotations.jaxrs.CookieParam`
+|`org.jboss.resteasy.reactive.RestCookie`
+|
+
+|`org.jboss.resteasy.annotations.jaxrs.MatrixParam`
+|`org.jboss.resteasy.reactive.RestMatrix`
+|
+
+|`org.jboss.resteasy.annotations.cache.Cache`
+|`org.jboss.resteasy.reactive.Cache`
+|
+
+|`org.jboss.resteasy.annotations.cache.NoCache`
+|`org.jboss.resteasy.reactive.NoCache`
+|
+
+|`org.jboss.resteasy.annotations.SseElementType`
+|`org.jboss.resteasy.reactive.RestStreamElementType`
+|
+
+|===
+
+=== JAX-RS providers
+
+Although RESTEasy Reactive provides the same spec compliant behavior as RESTEasy Classic does, it does not include the same exact provider implementations at runtime.
+
+The most common case where the difference in providers might result in different behavior, is the included `javax.ws.rs.ext.ExceptionMapper` implementations. To see what classes are included in the application, launch the application in dev mode and navigate to http://localhost:8080/q/dev/io.quarkus.quarkus-resteasy-reactive/exception-mappers.
+
+=== Multipart support
+
+HTTP Multipart support in RESTEasy Reactive does **not** reuse the same types or annotations as RESTEasy Classic and thus users are encouraged to read <<resteasy-reactive#multipart,this>> part of the reference documentation.
+
+=== Default media types
+
+Quarkus uses smart defaults when determining the media type of JAX-RS methods in order to simplify common use cases.
+The difference between `quarkus-resteasy-reactive` and `quarkus-resteasy` is the use of `text/plain` as the default media type instead of `text/html`
+when the method returns a `String`.
+
+== Client
+
+The Reactive REST Client (`quarkus-rest-client-reactive` and its dependencies) replace the legacy `quarkus-rest-client` but leverage Quarkus' build time processing
+and the unified I/O model provided by Vert.x.
+
```

Comment:
maybe https://quarkus.io/guides/rest-client-reactive#known-limitations is worth linking here too

Reply:
Feel free to add it :)